### PR TITLE
Fix engine luci auto rerun by setting logger

### PR DESCRIPTION
**Pull request by keyonghan**
This is a follow up of https://github.com/flutter/cocoon/pull/1072

Also increased the retry limit to 2.

Related issue: https://github.com/flutter/flutter/issues/68322

### CODE REVIEW

**Comment by iskakaushik** [comment]
Is there a sample task run I can look at? I'm curious how the logs look like.

**Comment by keyonghan** [comment]
[One sample log](https://pantheon.corp.google.com/logs/viewer?project=flutter-dashboard&resource=gae_app%2Fmodule_id%2Fdefault%2Fversion_id%2Fkeyonghantest2&minLogLevel=0&expandAll=false&timestamp=2021-02-03T19%3A03%3A23.528000000Z&customFacets&limitCustomFacetWidth=true&dateRangeStart=2021-02-02T19%3A03%3A23.780Z&dateRangeEnd=2021-02-03T19%3A03%3A23.780Z&interval=P1D&logName=projects%2Fflutter-dashboard%2Flogs%2Fappengine.googleapis.com%252Fstdout&logName=projects%2Fflutter-dashboard%2Flogs%2Fappengine.googleapis.com%252Fnginx.request&logName=projects%2Fflutter-dashboard%2Flogs%2Fappengine.googleapis.com%252Frequest_log&filters=text%3Apush-engine&scrollTimestamp=2021-02-02T22%3A41%3A06.156000000Z&advancedFilter=resource.type%3D%22gae_app%22%0Aresource.labels.module_id%3D%22default%22%0Aresource.labels.project_id%3D%22flutter-dashboard%22%0Aresource.labels.zone%3D%22us-central1-b%22%0Aresource.labels.version_id%3D%22keyonghantest2%22%0Atimestamp%3D%222021-02-02T22%3A41%3A06.156000000Z%22%0AinsertId%3D%221pvk2lag224w8wo%22)
Screen Shot 2021-02-03 at 11 07 21 AM
`retries: 0` means it will be retried.

**Comment by iskakaushik** [comment]
So these aren't visible to the end-users via the LUCI interface? I was hoping there would be an indication in the LUCI UI for the retries that are in-progress.

**Comment by keyonghan** [comment]
This is no direct indication whether a build is a rerun or not. But this can be verified on the builder page. For example: https://ci.chromium.org/p/flutter/builders/prod/Mac%20Web%20Engine, two rerun were triggered for commit: f94cf140c52c0cc3769d9281d5579e935886fbac
Screen Shot 2021-02-03 at 11 31 42 AM

File: app_dart/test/request_handlers/luci_status_test.dart [hunk 1]
```diff
@@ -47,7 +47,7 @@ void main() {
 
   setUp(() async {
     serviceAccountInfo = const ServiceAccountInfo(email: serviceAccountEmail);
-    config = FakeConfig(luciTryInfraFailureRetriesValue: 2, deviceLabServiceAccountValue: serviceAccountInfo);
+    config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
```
Comment (godofredoc) [inline app_dart/test/request_handlers/luci_status_test.dart:50]: Aren't we testing the retryCount anymore?

File: app_dart/test/service/luci_build_service_test.dart [hunk 2]
```diff
@@ -303,7 +303,6 @@ void main() {
       config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
       mockBuildBucketClient = MockBuildBucketClient();
       service = LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
-      config.luciTryInfraFailureRetriesValue = 3;
```
Comment (godofredoc) [inline app_dart/test/service/luci_build_service_test.dart:306]: Same here why are we removing the retries config?

File: app_dart/test/request_handlers/luci_status_test.dart (hunk 1 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/luci_status_test.dart:50]: We are. [`maxLuciTaskRetries`](https://devops-console-oss.corp.google.com/flutter/infra/+/master:config/devicelab_config.star) is the one we are using. `luciTryInfraFailureRetriesV` is some legacy parameter we have deprecated.

File: app_dart/test/service/luci_build_service_test.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/test/service/luci_build_service_test.dart:306]: Ditto.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**